### PR TITLE
bugfix of lua array

### PR DESCRIPTION
**Pull request by yqs112358**
help lua version of ScriptX to distinguish between so-called Array and common table

### CODE REVIEW

File: src/Reference.cc [hunk 1]
```diff
@@ -43,4 +43,8 @@ std::vector<std::string> Local<Object>::getKeyNames() const {
   return ret;
 }
 
+bool Local<Number>::isInteger() const {
+  return num.asNumber().toDouble() - num.asNumber().toInt64() < 1e-15;
```
Comment (LanderlYoung) [inline src/Reference.cc:47]: This is some magic code, Lua(also JS, etc) doesn't really provide Integer type.
If you really need to check this, do it in your on project code.

File: backend/Lua/LuaLocalReference.cc [hunk 2]
```diff
@@ -43,6 +43,29 @@ void ensureNonnull(int index) {
...
 
+bool judgeIsArray(int index)
+{
+  auto lua = currentLua();
+  int currectArrIndex = 0;
+  bool isArray = true;
+
+  lua_pushnil(lua); 
+  
+  while (lua_next(lua, index))
```
Comment (LanderlYoung) [inline backend/Lua/LuaLocalReference.cc:54]: In Lua you can't really check it's an Array or Object, and it's also documented clearly.
I don't think this check is good.

File: backend/Lua/LuaLocalReference.cc [hunk 3]
```diff
@@ -176,7 +202,11 @@ bool Local<Value>::isFunction() const {
   return val_ != 0 && lua_type(lua_backend::currentLua(), val_) == LUA_TFUNCTION;
 }
 
-bool Local<Value>::isArray() const { return isObject(); }
+bool Local<Value>::isArray() const {
```
Comment (LanderlYoung) [inline backend/Lua/LuaLocalReference.cc:205]: Also, this check may be called really often, adding heavy logic may affect the performance.

**Comment by LanderlYoung** [comment]
This is not a bug, and documented very cleared.

File: backend/Lua/LuaLocalReference.cc (hunk 3 above)
Comment (yqs112358) [reply backend/Lua/LuaLocalReference.cc:205]: Ok,,, it may be the truth 😂
Thanks for your advice, anyway

**Pull request closed without merging**